### PR TITLE
Confirm the user's already saved email

### PR DESCRIPTION
**Pull request by danroth27**
@HaoK Please review

### CODE REVIEW

**Comment by danroth27** [comment]
@phenning

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ public class InputModel
 
             var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
             var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
-            await _emailSender.SendEmailConfirmationAsync(Input.Email, callbackUrl);
+            await _emailSender.SendEmailConfirmationAsync(user.Email, callbackUrl);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Typically we don't access user's properties like this directly, instead to get the user's email, you go through identity: `await _userManager.GetEmailAsync(user)`

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Same issue exists for user.Id in the line above, that should probably be `await _userManager.GetUserIdAsync(user)`

Comment (danroth27) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: @DamianEdwards FYI - recommendation is to not access properties on ApplicationUser directly, but to go through UserManager instead.

**Comment by danroth27** [comment]
@HaoK @DamianEdwards @phenning Updated - please review.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs [hunk 2]
```diff
@@ -113,15 +116,17 @@ public class InputModel
...
+            var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
                 return RedirectToPage("/Error");
             }
 
             var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
-            var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
-            await _emailSender.SendEmailConfirmationAsync(Input.Email, callbackUrl);
+            var userId = await _userManager.GetUserIdAsync(user);
```
Comment (danroth27) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: @HaoK Would it be better to use the user manager to pull userId and username out of the principal? I.e. `_userManager.GetUserName(User)` and `_userManager.GetUserId(User)`

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Nah those are put in the principal on purpose as claims (NameIdentifier/Name by default).

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs [hunk 3]
```diff
@@ -53,18 +53,19 @@ public class InputModel
...
-            Username = user.UserName;
+            Username = await _userManager.GetUserNameAsync(user);
 
             Input = new InputModel()
             {
-                Email = user.Email,
-                PhoneNumber = user.PhoneNumber
+                Email = await _userManager.GetEmailAsync(user);
+                PhoneNumber = 
+                GetPhoneNumberAsync(user);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: should this be _userManager.GetPhoneNumberAsync?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs [hunk 4]
```diff
@@ -53,18 +53,19 @@ public class InputModel
...
         {
-            var user = await _userManager.GetUserAsync(HttpContext.User);
+            var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
                 return RedirectToPage("/Error");
             }
 
-            Username = user.UserName;
+            Username = await _userManager.GetUserNameAsync(user);
```
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Why do we need username on the model?  This should already be in the ClaimsPrincipal (User.Identity.Name) for the manage page, we aren't allowing this to be changed are we?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs (hunk 2 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: That said, the principal claims are configurable via: IdentityOptions.ClaimsIdentity, so maybe its a bit safer to go through the manager.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs (hunk 4 above)
Comment (danroth27) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: This page should reflect what is stored in the database. The username is not editable. It's just displayed as part of the user profile.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs (hunk 2 above)
Comment (danroth27) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: I assume the user ID never changes, so I think calling `_userManager.GetUserId(User)` should be fine. For the username, I think it really depends if you are ok with username stored in the cookie, or if you want the most up to date username from the identity store. @HaoK Does that sound right?

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Yeah its not a big deal either way, you'd assume any app that let them change the username, would also clear the cookie/sign them in again as part of the change name flow.  That's what our manage actions do already...

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs (hunk 4 above)
Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Yeah so if its read only data on the ManagePage, you probably could just use User.Identity.Name in the view in this case.

Comment (HaoK) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs]: Yeah that's what we do already for the "Hello, DanRoth" when you are signed in:

That view does: Hello, @User.Identity.Name!

No reason not to do the same

**Comment by danroth27** [comment]
@DamianEdwards is handling these changes as part of https://github.com/dotnet/templating/pull/1070

**Pull request closed without merging**